engine-precompiles/src/native.rs

```diff
@@ -537,7 +533,7 @@ impl<I: IO> Precompile for ExitToEthereum<I> {
                     .try_into()
                     .map_err(|_| ExitError::Other(Cow::from("ERR_INVALID_RECIPIENT_ADDRESS")))?;
                 (
-                    self.current_account_id.clone(),
+                    get_eth_connector_contract_account(&self.io)?,
                     // There is no way to inject json, given the encoding of both arguments
                     // as decimal and hexadecimal respectively.
                     WithdrawCallArgs {
```

Comment:
One more change is needed here. As @karim-en pointed out in our call, the Engine Withdraw used Borsh serialization for the arguments and [so does the new eth-connector](https://github.com/aurora-is-near/aurora-eth-connector/blob/master/eth-connector/src/lib.rs#L488), however regular NEP-141 tokens use JSON (see ERC-20 implementation below).

I think there are two options for what to do:

1. Change the eth-connector to use JSON args for withdraw, thus making it consistent with the other bridged NEP-141 tokens, and then make the corresponding change here.
2. Add another config to the Engine to say what format the withdraw arguments are supposed to be in.

I personally think 1 is the better solution.

Reply:
I also like the first option, but this change should be backward compatible, because this change can break any services or contracts that interact with the engine contract.
To make it backward compatible I think we can try to deserialise the input twice, as an borsh, and then try to deserialise it as a json if it was failed.

Reply:
@birchmd what existing problem or business goal is it to solve by introducing changes that break the existing API? I do not have enough context to understand the essence of such a controversial proposal.

Reply:
@karim-en, what is the JSON format for this message should be?

Reply:
@karim-en Backwards compatibility is maintained by the main Aurora Engine `withdraw` (which delegates to the new connector's `engine_withdraw`) remaining borsh-encoded. I don't think we need fallback serialization on the new `withdraw` method of the new connector.

@aleksuss the format will be identical to [the ERC-20 case](https://github.com/aurora-is-near/aurora-engine/blob/77c284a5c231f84c1ec4c2080841528170e0413c/engine-precompiles/src/native.rs#L593).

@mrLSD This is not a breaking change to the existing API because the Aurora Engine maintains backwards compatibility. The reason to API of the new contract to be different from Aurora Engine is because then it is consistent with all other bridged NEP-141 tokens (i.e. all ERC-20 tokens bridged from Ethereum). All NEP-141 tokens having a consistent interface allows us to use any bridged token as the base token for a silo, not just ETH. Having alternate base tokens is a business requirement.

Reply:
@birchmd don't we need to support both versions of deserialization in the main aurora contract in the `withdraw` function? In case we will use the main contract as an eth-connector for using ETH as the base token in silo?